extensions/vault/runtime/src/main/java/io/quarkus/vault/health/VaultHealthCheck.java

```diff
@@ -0,0 +1,57 @@
+package io.quarkus.vault.health;
+
+import io.quarkus.vault.VaultSystemBackendEngine;
+import io.quarkus.vault.runtime.sys.VaultHealth;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Readiness;
+
+@Readiness
+@ApplicationScoped
+public class VaultHealthCheck implements HealthCheck {
+
+    @Inject
+    VaultSystemBackendEngine vaultSystemBackendEngine;
+
+    @Override
+    public HealthCheckResponse call() {
+
+        final HealthCheckResponseBuilder builder = HealthCheckResponse.named("Vault connection health check");
+
+        try {
+            final VaultHealth vaultHealth = this.vaultSystemBackendEngine.health(false, false);
```

Comment:
The standby and perf customizations are not possible here

Reply:
Not here, but if they are configured in `application.properties`, then these values are overridden. I know that you are asking why are you leaving this instead of provide a `health()` method. The reason is that `VaultSystemBackendEngine` could be injected in any class and users programmatically might want to set them programmatically.

Reply:
@lordofthejars thanks, but then we'd support the users overwriting the configuration, and when the config is good, then they'd have to type `false, false` :-)
Having `health()` alone is quite enough IMHO, it does seem to me in most cases health of the running active Vault would be of interest. `health(boolean standbyok)` and `health(boolean perfok)` can be easily added if someone ever asks :-)